distributed/diagnostics/tests/test_install_plugin.py

```diff
@@ -28,9 +28,9 @@ async def test_pip_install(c, s, a):
                 PipInstall(packages=["requests"], pip_options=["--upgrade"])
             )
             assert Popen.call_count > 0
-            args = Popen.call_args[0][0]
-            assert "python" in args[0]
-            assert args[1:] == ["-m", "pip", "install", "--upgrade", "requests"]
+            python, *args, requirements_file = Popen.call_args[0][0]
+            assert "python" in python
+            assert args == ["-m", "pip", "install", "--upgrade", "-r"]
```

Comment:
I get that it's difficult to test but I would add a comment here why the `-r` is important. From first principle, this test is otherwise pretty useless. We test that pip install something but we also care that it is using the requirements file mechanism since this comes with additional features.
Adding a comment elevates this from a too detailed unit test that asserts on an implementation detail to an actual intended feature

Reply:
Fair point!

Reply:
Done